R/source.R

```diff
@@ -36,7 +36,15 @@ source_file <- function(path, env = test_env(), chdir = TRUE,
   if (wrap) {
     invisible(test_code(NULL, exprs, env))
   } else {
-    invisible(eval(exprs, env))
+    tryCatch(
```

Comment:
To allow recover and deep backtraces:

```suggestion
    withCallingHandlers(
```